### DISCUSSION TITLE
WIFI Connection Fails

### DISCUSSION THREAD

**Discussion by brannon-zahand** [iRobotEducation/create3_docs#391] [Questions]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Ubuntu 20.04.06 running in VirtualBox on Windows 11

### Which version of ROS 2 is installed on your computer?

Galactic

### Which firmware version is installed on your robot?

G.4.5

### Which RMW is your robot running?

FastRTPS

### Does your robot have an assigned namespace? If so, please share.

/robot

### Is the robot connected to a network? If so, what is the network type?

Yes, home network

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

Yes

### What is the Adapter Board's USB/BLE Toggle currently switched to?

Bluetooth (default)

### Describe your question.

Can't seem to connect to my Create3 via WIFI.  When I do a "ros2 topic list" I only get: 

     _/parameter_events
     /rosout_

When I try to send a command to undock I get:  _Waiting for an action server to become available..._

Confirmed my ASUS router has multicast enabled. That said, I tried with both Cyclone and FastRTSP as well as multicast on/off with FASTRTSP using a unicast_locator.xml file, but no luck. Also, confirmed that I only have a single (RMW_IMPLEMENTATION) in my .bashrc file that is using FASTRTPS.

Also tried this using an Ubuntu instance with WSL and had the exact same problem. Also tried with an entirely different wifi router... no luck as well.

Bluetooth I can connect to the Create3 fine, and similarly I can connect to the Create3 through a web browser and change its settings, etc.  I can also successfully ping the Create3 from the Ubuntu VM.

Any help would be greatly appreciated!  I've tried the steps others have in previous discussions with no luck. :(

**Comment by kwujciak** [comment]
In my experience, I found more luck using VMware as opposed to Virtual Box. I see you've already tried this, but I've had the most consistent connection while using FastRTPS and disabling multicast when using a VM (make sure you'll using the correct IP in your .xml). I'm assuming you're getting adequate wifi connection through your VM since you were able to ping your robot. What IP are you pinging? The first thing I would check is that you have strong and consistent wifi on your VM. If you're not getting wifi on your VM, a network adapter could solve that problem. If you try to disable multicast, I would also remind you that you need to include that in your bash script with a line such as:

`export FASTRTPS_DEFAULT_PROFILES_FILE='/home/{your user}/{location of file}/disable_multicast_example.xml'
`

Once you have added the implementation line and disable multicast line in your bash, try these commands:

```
ros2 daemon stop
source ~/.bashrc
ros2 topic list
```

Let me know if this helps.

**Reply by brannon-zahand** [reply]
Thanks for the tip on VMware, will try that using FASTRTPS and see what happens. 

As an aside, I got the robot working on a laptop running Ubuntu 20.04 over my wifi network, so the issue seems to be something to do with the connection between the virtual machine and my network, not my network or the robot.

**Reply by kwujciak** [reply]
If that's the case, then you'll likely need a wifi dongle. I had the same issue where I couldn't get my wifi to bridge to my VM. To remedy this, I used a wifi dongle and that made all the difference. Go to [this page](https://iroboteducation.github.io/create3_docs/setup/network-config/) and scroll down to the virtual machine section (look at the option with the network adapter).
Let me know if that works!
